dataset/classification_datasets/graph_dataset.py

```diff
@@ -0,0 +1,70 @@
+import json
+from os import listdir
+from os.path import join, dirname, isdir
+
+import torch
+from omegaconf import DictConfig
+from torch_geometric.data import InMemoryDataset, Data
+from tqdm import tqdm
+
+from utils import is_json_file
+
+
+class GraphDataset(InMemoryDataset):
+    def __init__(self, config: DictConfig, stage: str, transform=None, pre_transform=None):
+        self.root = join(config.data_folder, config.dataset.name, config.dataset.dir, stage)
+        self.processed_file_path = f"{join(self.root, stage)}.ptg"
+
+        vocab_path = join(config.data_folder, config.dataset.name, config.dataset.dir, config.dataset.vocab_file)
+        with open(vocab_path, "r") as f:
+            vocab = json.load(f)
+        self.v_type2id = vocab["v_type2id"]
+        self.v_name2id = vocab["v_name2id"]
+        self.e_type2id = vocab["e_type2id"]
+
+        super(GraphDataset, self).__init__(self.root, transform, pre_transform)
+
+        self.data, self.slices = torch.load(self.processed_file_path)
+
+    @property
+    def raw_file_names(self):
```

Comment:
It's not good to make property complicated function. Especially if it calls too often. I suggest storing the result in inner variable and returns it if it already calculated.